### DISCUSSION TITLE
现在runner.get_column_lineage()的结果是排序的。如何能得到insert的原始字段顺序？

### DISCUSSION THREAD

**Discussion by jincuodao9903** [reata/sqllineage#664] [Q&A]
hive数据库用insert overwrite写入的时候，不会匹配字段名，而是按照insert的字段顺序写入。现在runner.get_column_lineage返回的结果没有了原始insert的字段顺序，导致血缘字段和hive数据模型的字段无法匹配。
有什么办法可以取到SQL脚本中insert的原始字段顺序？

**Comment by kingbaiyulong** [comment]
We encountered the same problem. The SQL statement(hive sql) is as follows:
INSERT OVERWRITE table data_ds.data_metric_di PARTITION(ds=20251013)
SELECT
uin,
device,
play_cnt,
play_progress90_cnt
FROM data_dwd.data_metric_ori
where ds=20251013
The actual fields of the table data_ds.data_metric_di in this SQL statement are [uin, device, pcr_cnt, and pcr90_cnt.] The column lineage   returned by SQLLineage does not match the actual order. This problem could be resolved if SQLLineage supported returning field order.
